core/src/test/java/org/apache/iceberg/hadoop/HadoopFileIOTest.java

```diff
@@ -166,6 +170,8 @@ public void testHadoopFileIOJavaSerialization() throws IOException, ClassNotFoun
 
   private List<Path> createRandomFiles(Path parent, int count) {
     List<Path> paths = Lists.newArrayList();
```

Comment:
I think the underlying error here is using an arrayList in a multi threaded context. I switched this with a Vector implementation and so far have run the "testDeleteFiles" test 12,000 times without error.

Reply:
we could also try and remove the `parallel()` during random file creation. That should resolve the issue as well

Reply:
@danielcweeks thoughts on removing `parallel()` from `createRandomFiles()`?

Reply:
So my final thoughts here are we should switch this to Tasks.foreach with a dedicated executor service and switch the implementation here to use a thread safe collection like vector.

Reply:
![image](https://user-images.githubusercontent.com/413025/229888065-056c1736-d37b-4f5c-bb27-7656e424e64b.png)
Up to 75K runs without issue, Going to have to stop soon as IntelliJ UI memory usage is gonna be a limiting factor soon

Reply:
+1 for Russell's suggestion